src/Form/FieldTextarea.js

```diff
@@ -5,30 +5,94 @@ import React from 'react';
 
 import FieldInput from './FieldInput';
 import IconEdit from './icons/IconEdit';
+import Util from '../Util/Util';
+
+class FieldTextarea extends FieldInput {
+  constructor() {
+    super(...arguments);
+
+    this.state = {height: 0};
+
+    this.updateTextareaHeight = Util.throttle(this.updateTextareaHeight, 100);
+    this.handleContentEditableBlur = this.handleContentEditableBlur.bind(this);
+    this.handleContentEditableChange =
+      this.handleContentEditableChange.bind(this);
+    this.handleContentEditableFocus =
+      this.handleContentEditableFocus.bind(this);
+  }
+
+  componentWillMount() {
+    this.setState({height: this.props.minHeight});
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+
+    if (this.isEditing() || this.props.writeType === 'input') {
+      this.updateTextareaHeight(this.refs.inputElement);
+
+      // React throws a warning if children are specified in an element with
+      // contenteditable="true", so this hack allows us to set a default value
+      // for this form field.
+      if (this.props.startValue) {
+        this.refs.inputElement.textContent = this.props.startValue;
+      }
+    }
+  }
+
+  handleContentEditableBlur(event) {
+    let {props} = this;
+    props.handleEvent('blur', props.name, event.target.innerText, event);
+  }
+
+  handleContentEditableChange(event) {
+    let {props} = this;
+
+    this.updateTextareaHeight(event.target);
+    props.handleEvent('change', props.name, event.target.innerText, event);
+  }
+
+  handleContentEditableFocus(event) {
+    let {props} = this;
+    props.handleEvent('focus', props.name, event.target.innerText, event);
```

Comment:
We have to "proxy" `handleEvent` because `FieldInput#handleEvent` looks at `event.target.value`, which is `undefined` for `contentEditable` elements. Instead we provide `innerText` which contains text (with line breaks and other whitespace characters).